enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,320 @@
+The functionality of some cluster operators is fully replaced by the logic in the hosted control plane (e.g. authentication operator).
+The ingress operator is an example of a cluster operator whose life cycle is shared between the cluster version operator and HyperShift.
+HyperShift doesn't pass all the ingress operator's payload to the cluster version operator.
+Certain manifests are filtered out, most importantly: [the ingress operator's deployment](https://github.com/openshift/hypershift/blob/0a5ec529ce7ae6f238ed27be6eb5757cf70c41f9/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L75).
+The reason for this is that the ingress operator is considered as a part of the hosted control plane hence deployed on the control plane cluster.
+However [the ingress operator's APIs](#ingress-operator-apis) and some RBACs to manage those APIs are still kept in the cluster version operator's payload.
+The ingress capability comes handy here as HyperShift can leverage it to control the remaining (not filtered) payload.
+
+The HyperShift's API which will handle the hosted cluster capabilities is out of the scope of this enhancement. However this enhancement is built with the assumption that the `ClusterVersion`'s capability API (spec and status) is sufficient for any capability implementation in HyperShift.
+
+#### Standalone OpenShift
+
+Making the ingress operator optional for the standalone OpenShift is a bigger scope than for HyperShift.
+The changes impact the components which are not used (or used differently) in HyperShift like the installer, the authentication and the monitoring operators.
+To prevent the scope creep for [RFE-3395](https://issues.redhat.com/browse/RFE-3395), the following approach is proposed:
+- the cluster version operator should be able to add required capabilities.
+- the required capabilities is a configuration option of the cluster version operator (e.g. `--required-capabilities` flag).
+- the cluster version operator should [implicitly enable](https://github.com/openshift/cluster-version-operator/blob/2fd54d18de83b617d9d30c28cfee6383e430a102/pkg/cvo/status.go#L161-L165) all the provided required capabilities so that they appear in the status of `ClusterVersion` object.
+- the required capabilities should be added to [the cluster version operator manifests](https://github.com/openshift/cluster-version-operator/blob/master/install/0000_00_cluster-version-operator_03_deployment.yaml) only.
+- the ingress capability should be required.
+
+This approach should make any required capability read only on the standalone OpenShift.
+Also, it doesn't let capabilities be set as required on HyperShift because HyperShift doesn't use the cluster version operator's deployment manifest but [manages the operator's deployment directly](https://github.com/openshift/hypershift/blob/15fe51bda33b7acede2b4b7a7c66acc2f369a8a2/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L318),
```

Comment:
* Hyphenate the compound adjective "read-only" to disambiguate it.  
* Use "cause" instead of "let" (enabling the required capability is still permitted and can be caused using a different API on HyperShift).  
```suggestion
This approach should make any required capability read-only on standalone OpenShift.
Also, it doesn't cause the required capabilities to be set on HyperShift because HyperShift doesn't use the cluster version operator's deployment manifest but [manages the operator's deployment directly](https://github.com/openshift/hypershift/blob/15fe51bda33b7acede2b4b7a7c66acc2f369a8a2/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L318),
```